### PR TITLE
refactor: tencent cos source & scf sink

### PR DESCRIPTION
**Pull request by wenfengwang**
**Changes:**
- refactor Tencent cos source 
- refactor Tencent scf sink

### CODE REVIEW

File: connectors/source-tencentcloud-cos/internal/server.go [hunk 1]
```diff
@@ -51,15 +53,21 @@ var (
...
+	config.SourceConfig
 	B        Bucket   `json:"bucket" yaml:"bucket"`
 	F        Function `json:"function" yaml:"function"`
 	Debug    bool     `json:"debug" yaml:"debug"`
 	Eventbus string   `json:"eventbus" yaml:"eventbus"`
 	Secret   *Secret  `json:"-" yaml:"-"`
 }
 
+func (c *Config) GetSecret() config.SecretAccessor {
+	return c.Secret
```
Comment (xdlbdy) [inline connectors/source-tencentcloud-cos/internal/server.go:68]: `secret` is pointer, it zero value is nil ,can not set value

File: connectors/sink-tencentcloud-scf/internal/server.go [hunk 2]
```diff
@@ -33,19 +32,26 @@ import (
...
+
 type Config struct {
+	config.SinkConfig
 	F      Function `json:"function" yaml:"function"`
 	Debug  bool     `json:"debug" yaml:"debug"`
 	Secret *Secret  `json:"-" yaml:"-"`
 }
 
+func (c Config) GetSecret() config.SecretAccessor {
+	return c.Secret
```
Comment (xdlbdy) [inline connectors/sink-tencentcloud-scf/internal/server.go:52]: secret is pointer, it zero value is nil ,can not set value

File: connectors/sink-tencentcloud-scf/internal/server.go [hunk 3]
```diff
@@ -33,19 +32,26 @@ import (
...
 )
 
 var (
 	functionNamePrefix = "vanus-cos-source-function"
 )
 
+var _ config.SinkConfigAccessor = &Config{}
+
 type Config struct {
+	config.SinkConfig
```
Comment (xdlbdy) [inline connectors/sink-tencentcloud-scf/internal/server.go:45]:
```suggestion
	cdkgo.SinkConfig
```
cdkgo has some alias, you can reference it

File: connectors/sink-tencentcloud-scf/internal/server.go [hunk 4]
```diff
@@ -65,56 +71,56 @@ func NewFunctionSink() connector.Sink {
...
-	})
-	return protocol.ResultACK
-}
 
-func (c *functionSink) Init(cfgPath, secretPath string) error {
-	cfg := &Config{}
-	if err := cdkutil.ParseConfig(cfgPath, cfg); err != nil {
-		return err
-	}
+	return connector.Success
```
Comment (xdlbdy) [inline connectors/sink-tencentcloud-scf/internal/server.go:104]:
```suggestion
	return cdkgo.Success
```

File: connectors/sink-tencentcloud-scf/internal/server.go [hunk 5]
```diff
@@ -33,19 +32,26 @@ import (
...
+var _ config.SinkConfigAccessor = &Config{}
+
 type Config struct {
+	config.SinkConfig
 	F      Function `json:"function" yaml:"function"`
 	Debug  bool     `json:"debug" yaml:"debug"`
 	Secret *Secret  `json:"-" yaml:"-"`
 }
 
+func (c Config) GetSecret() config.SecretAccessor {
```
Comment (xdlbdy) [inline connectors/sink-tencentcloud-scf/internal/server.go:51]:
```suggestion
func (c Config) GetSecret() cdkgo.SecretAccessor {
```

File: connectors/sink-tencentcloud-scf/internal/server.go [hunk 6]
```diff
@@ -65,56 +71,56 @@ func NewFunctionSink() connector.Sink {
...
+
 type functionSink struct {
 	scfClient *v20180416.Client
-	logger    log.Logger
 	cfg       *Config
 	funcName  string
 }
 
-func (c *functionSink) Receive(_ context.Context, event v2.Event) protocol.Result {
+func (c *functionSink) Arrived(_ context.Context, events ...*v2.Event) connector.Result {
```
Comment (xdlbdy) [inline connectors/sink-tencentcloud-scf/internal/server.go:82]:
```suggestion
func (c *functionSink) Arrived(_ context.Context, events ...*v2.Event) cdkgo.Result {
```

File: connectors/sink-tencentcloud-scf/internal/server.go [hunk 7]
```diff
@@ -65,56 +71,56 @@ func NewFunctionSink() connector.Sink {
...
+		e := events[idx]
+		payload := string(e.Data())
+		req.ClientContext = &payload
+
+		res, err := c.scfClient.Invoke(req)
+		if err != nil {
+			log.Debug("failed to invoke function", map[string]interface{}{
+				log.KeyError: err,
+			})
+			return connector.NewResult(http.StatusInternalServerError, err.Error())
```
Comment (xdlbdy) [inline connectors/sink-tencentcloud-scf/internal/server.go:97]:
```suggestion
			return cdkgo.NewResult(http.StatusInternalServerError, err.Error())
```

File: connectors/sink-tencentcloud-scf/internal/server.go [hunk 8]
```diff
@@ -65,56 +71,56 @@ func NewFunctionSink() connector.Sink {
...
-	cfg := &Config{}
-	if err := cdkutil.ParseConfig(cfgPath, cfg); err != nil {
-		return err
-	}
+	return connector.Success
+}
 
-	if cfg.F.Namespace == "" {
-		cfg.F.Namespace = "default"
+func (c *functionSink) Initialize(_ context.Context, cfg config.ConfigAccessor) error {
```
Comment (xdlbdy) [inline connectors/sink-tencentcloud-scf/internal/server.go:107]:
```suggestion
func (c *functionSink) Initialize(_ context.Context, cfg cdkgo.ConfigAccessor) error {
```

File: connectors/source-tencentcloud-cos/internal/server.go (hunk 1 above)
Comment (wenfengwang) [reply connectors/source-tencentcloud-cos/internal/server.go:69]: DONE

File: connectors/sink-tencentcloud-scf/internal/server.go (hunk 2 above)
Comment (wenfengwang) [reply connectors/sink-tencentcloud-scf/internal/server.go:53]: DONE

File: connectors/sink-tencentcloud-scf/internal/server.go (hunk 4 above)
Comment (wenfengwang) [reply connectors/sink-tencentcloud-scf/internal/server.go:111]: `Success` no in aliases

Comment (xdlbdy) [reply connectors/sink-tencentcloud-scf/internal/server.go:111]: I forget it, it is `SuccessResult` not `Success`

**Review by xdlbdy — APPROVED** [review]
LGTM

**Pull request merged**